File: baekjoon/18110.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <cmath>
using namespace std;

int main() {
    ios_base::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);

    int n;
    cin >> n;

    if (n == 0) {
        cout << 0 << '\n';
        return 0;
    }

    vector<int> a(n);
    for (int i = 0; i < n; i++) cin >> a[i];

    sort(a.begin(), a.end());

    int ex = floor(n * 0.15 + 0.5);
    int sum = 0;

    for (int i = ex; i < n - ex; i++) {
        sum += a[i];
    }

    double avg = double(sum) / (n - 2 * ex);

    cout << floor(avg + 0.5) << '\n';

    return 0;
}

File: baekjoon/18110.py
from sys import stdin
import math

n = int(stdin.readline())
data = []

if n == 0:
    print(0)
    exit()

for i in range(0, n):
    data.append(int(stdin.readline()))

data.sort()

exc = math.floor(n * 0.15 + 0.5)
sum = 0

for i in range(exc, n-exc):
    sum += data[i]

avg = sum / (n - 2*exc)

print(math.floor(avg + 0.5))